camshare/camshare/ViewControllers/TipsAndTricsViewController.swift

```diff
@@ -76,11 +69,168 @@ class TipsAndTricsViewController: UIViewController, UITableViewDelegate, UITable
-        cell.statusSegmentControl.addTarget(self, action: #selector(statusChanged), for: .valueChanged)
+
+        if selectedSegment == 0 {
+            cell.cellView.isHidden = false
+            cell.headingLabel.text = tipsAndTricksContent[indexPath.item].heading
+            cell.bodyTextView.text = tipsAndTricksContent[indexPath.item].body
+
+            let currentRatingTap = UITapGestureRecognizer(target: self, action: #selector(currentRatingTapped(_:)))
+            cell.currentRatingImageView.isUserInteractionEnabled = true
+            cell.currentRatingImageView.tag = indexPath.row
+            cell.currentRatingImageView.addGestureRecognizer(currentRatingTap)
+        }
+
+        if selectedSegment == 1 {
+            cell.cellView.isHidden = true
+        }
+
         return cell
     }
-    
+
+    @IBAction func currentRatingTapped(_ sender: AnyObject) {
+        print("Tap Gesture for Cell Works: tapped - \(sender.view.tag)")
+        let indexPath = NSIndexPath(row: sender.view.tag, section: 0)
+        if let cell = tableView.cellForRow(at: indexPath as IndexPath) {
+            //swiftlint:disable all
+            let currentCell = cell as! TipsAndTricsTableViewCell
+            //swiftlint:enable all
+            currentCell.animateRatings()
+
+            let veryBadRatingTap = UITapGestureRecognizer(target: self, action: #selector(veryBadRatingTapped(_:)))
+            let badRatingTap = UITapGestureRecognizer(target: self, action: #selector(badRatingTapped(_:)))
+            let okayRatingTap = UITapGestureRecognizer(target: self, action: #selector(okayRatingTapped(_:)))
+            let goodRatingTap = UITapGestureRecognizer(target: self, action: #selector(goodRatingTapped(_:)))
+            let veryGoodRatingTap = UITapGestureRecognizer(target: self, action: #selector(veryGoodRatingTapped(_:)))
+            currentCell.veryBadRatingImageView.addGestureRecognizer(veryBadRatingTap)
+            currentCell.badRatingImageView.addGestureRecognizer(badRatingTap)
+            currentCell.okayRatingImageView.addGestureRecognizer(okayRatingTap)
+            currentCell.goodRatingImageView.addGestureRecognizer(goodRatingTap)
+            currentCell.veryGoodRatingImageView.addGestureRecognizer(veryGoodRatingTap)
+            currentCell.veryBadRatingImageView.tag = indexPath.row
+            currentCell.badRatingImageView.tag = indexPath.row
+            currentCell.okayRatingImageView.tag = indexPath.row
+            currentCell.goodRatingImageView.tag = indexPath.row
+            currentCell.veryGoodRatingImageView.tag = indexPath.row
+            currentCell.veryBadRatingImageView.isUserInteractionEnabled = true
+            currentCell.badRatingImageView.isUserInteractionEnabled = true
+            currentCell.okayRatingImageView.isUserInteractionEnabled = true
+            currentCell.goodRatingImageView.isUserInteractionEnabled = true
+            currentCell.veryGoodRatingImageView.isUserInteractionEnabled = true
```

Comment:
You can use an IBCollection in the currentCell subsection and add this in a for loop, I would suggest using NIBs for table view cells helps clean up your code.

Reply:
Will check it out thanks